### PR TITLE
Fix CI badges in Readme.md

### LINKED ISSUE #51005: README Azure badges shows as unknown
image

The relevant script for generating this part of README:

https://github.com/dotnet/roslyn/blob/master/scripts/generate-badges.ps1

### PR DESCRIPTION
**Pull request by MaStr11**
Fix #51005

A gist with the new badges can be found here:

https://gist.github.com/MaStr11/ccfa7ca083457bcb75a09d5c5da49997

### CODE REVIEW

**Comment by MaStr11** [comment]
The following jobs are shown/missing

###  roslyn-CI
* [ ] roslyn-CI (Build_Unix_Debug)
* [ ] roslyn-CI (Build_Windows_Debug)
* [ ] roslyn-CI (Build_Windows_Release)
* [x] roslyn-CI (Correctness_Build)
* [x] roslyn-CI (Correctness_Determinism)
* [ ] roslyn-CI (Correctness_SourceBuild)
* [x] roslyn-CI (Test_Linux_Debug)
* [ ] roslyn-CI (Test_Linux_Debug_Single_Machine)
* [x] roslyn-CI (Test_Windows_CoreClr_Debug)
* [ ] roslyn-CI (Test_Windows_CoreClr_Debug_Single_Machine)
* [x] roslyn-CI (Test_Windows_CoreClr_Release)
* [x] rosln-CI (Test_Windows_Desktop_Debug_32)
* [x] roslyn-CI (Test_Windows_Desktop_Debug_64)
* [x] roslyn-CI (Test_Windows_Desktop_Release_32)
* [x] roslyn-CI (Test_Windows_Desktop_Release_64)
* [x] roslyn-CI (Test_Windows_Desktop_Spanish_Release_32)
* [x] roslyn-CI (Test_macOS_Debug)
### roslyn-integration-CI
* [x] roslyn-integration-CI (VS_Integration debug_32)
* [x] roslyn-integration-CI (VS_Integration debug_64)
* [x] roslyn-integration-CI (VS_Integration release_32)
* [x] roslyn-integration-CI (VS_Integration release_64)

File: scripts/generate-badges.ps1 [hunk 1]
```diff
@@ -76,15 +76,15 @@ function Get-CoreClrTable() {
...
+        'VS_Integration#debug_32',
+        'VS_Integration#debug_64',
+        'VS_Integration#release_32',
+        'VS_Integration#release_64'
     )
 
     $table = @'
 ### Integration Tests
-|Branch|Debug|Release|Debug (Legacy completion)|Release (Legacy completion)
+|Branch|Debug x86|Debug x64|Release x86|Release x64
```
Comment (MaStr11) [inline scripts/generate-badges.ps1:87]: I changed the order to be in line with *Desktop* ordering.

File: scripts/generate-badges.ps1 [hunk 2]
```diff
@@ -98,15 +98,15 @@ function Get-IntegrationTable() {
 
 function Get-MiscTable() {
     $jobNames = @(
-        'Windows_Determinism_Test',
-        'Windows_Correctness_Test',
-        'Windows_Desktop_Spanish_Unit_Tests',
-        'Linux_Test#mono'
+        'Correctness_Determinism',
+        'Correctness_Build',
```
Comment (MaStr11) [inline scripts/generate-badges.ps1:102]: There is also `Correctness_SourceBuild`. Should it be integrated?

File: scripts/generate-badges.ps1 [hunk 3]
```diff
@@ -98,15 +98,15 @@ function Get-IntegrationTable() {
...
+        'Test_Windows_Desktop_Spanish_Release_32',
+        'Test_macOS_Debug'
     )
 
     $table = @'
 ### Misc Tests
-|Branch|Determinism|Build Correctness|Mono|Spanish|
+|Branch|Determinism|Build Correctness|Spanish|MacOS|
```
Comment (MaStr11) [inline scripts/generate-badges.ps1:109]: The header ordering was wrong and "Linux#mono" was replaced by "MacOS".

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
I think you need to run the script to get the README updated.

**Comment by MaStr11** [comment]
The PR is a work in progress until https://github.com/dotnet/roslyn/pull/51085#issuecomment-775875319 is addressed. Until then I use the [gist](https://gist.github.com/MaStr11/ccfa7ca083457bcb75a09d5c5da49997) from the PR description, which allows copy/paste without fiddling with the readme.

**Comment by RikkiGibson** [comment]
Thank you for your contribution!

> According to [#51005 (comment)](https://github.com/dotnet/roslyn/issues/51005#issuecomment-775466332) the "Build.." jobs should be added. Should these get a separate paragraph?

Yeah let's put the Build jobs in their own paragraph.

> What about the "Single_Machine" jobs?

Let's not include these. They're more about validating a code path in our infrastructure than about validating the product.

> What about "Correctness_SourceBuild"?

Yeah, let's include this after the "Build Correctness" column.

**Comment by MaStr11** [comment]
@RikkiGibson The changed overview (including a new "Builds" paragraph) can be found in the gist https://gist.github.com/MaStr11/ccfa7ca083457bcb75a09d5c5da49997

File: scripts/generate-badges.ps1 [hunk 4]
```diff
@@ -34,6 +34,26 @@ function Get-AzureLine($branchName, $jobNames, [switch]$integration = $false) {
...
+function Get-BuildsTable() {
+    $jobNames = @(
+        'Build_Windows_Debug'
+        'Build_Windows_Release'
+        'Build_Unix_Debug'
+    )
+
+    $table = @'
+### Builds
+|Branch|Windows Debug|Windows Release|Unix Debug|
```
Comment (Youssef1313) [inline scripts/generate-badges.ps1:46]: nit: Some markdown parsers (e.g. kramdown) require a blank line around headings. See [MD022](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md022---headings-should-be-surrounded-by-blank-lines).

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
LGTM pending update of the content in-repo :)

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
@Youssef1313 did the author address all your concerns?

**Review by Youssef1313 — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Merging with single sign-off as this is a documentation-only change.

**Comment by RikkiGibson** [comment]
Thanks for your contribution @MaStr11!

**Pull request merged**